addons/im_livechat/static/src/core/web/thread_patch.js

```diff
@@ -0,0 +1,44 @@
+import { Thread } from "@mail/core/common/thread";
+import { _t } from "@web/core/l10n/translation";
+import { user } from "@web/core/user";
+import { patch } from "@web/core/utils/patch";
+
+const { DateTime } = luxon;
+
+patch(Thread.prototype, {
+    get livechatVisitor() {
+        return this.props.thread.livechatVisitorMember;
+    },
+    get showVisitorDisconnected() {
+        return (
+            this.props.thread.channel_type === "livechat" &&
+            this.store.self.isInternalUser &&
+            this.props.thread.livechat_active &&
+            this.livechatVisitor &&
+            this.livechatVisitor.persona.im_status != "online"
+        );
+    },
+    get disconnectedText() {
+        const lastPoll = this.livechatVisitor.persona.last_poll;
+        if (!lastPoll) {
+            return _t("Visitor is disconnected");
+        }
+        const userLocale = { locale: user.lang };
+        if (lastPoll.hasSame(DateTime.now(), "day")) {
+            return _t("Visitor is disconnected since Today at %(time)s", {
+                time: lastPoll.toLocaleString(DateTime.TIME_24_SIMPLE, userLocale),
+            });
+        }
+        if (lastPoll.hasSame(DateTime.now().minus({ day: 1 }), "day")) {
+            return _t("Visitor is disconnected since Yesterday at %(time)s", {
+                time: lastPoll.toLocaleString(DateTime.TIME_24_SIMPLE, userLocale),
+            });
+        }
+        return _t("Visitor is disconnected since %(time)s", {
+            time: lastPoll.toLocaleString(
+                { ...DateTime.DATETIME_MED, hourCycle: "h23", year: undefined },
+                userLocale
+            ),
+        });
+    },
```

Comment:
Could you write some small tests to cover the feature and the various possible displays?